truss-build/src/truss/build/image.py

```diff
@@ -0,0 +1,165 @@
+from .commands import *
+from typing import Dict, List, Optional, Union
+from pathlib import Path
+from .utils import flatten_str_args, make_pip_install_args
+import shlex
+
+# TODO(bola): support secrets
+
+
+class Image:
+    def __init__(self) -> None:
+        self._base_image = ""
+        self._commands: list[Command] = []
+        self._base_image: Optional[str] = None
+
+    # TODO(support static constructors for different ways of pulling base iamge)
+    # Image.from_dockerfile, from_aws_ecr, from_gcp_artifact_registry, from_registry
+    # also can add common base images: Image.truss_base()....
+
+    def serialize(self) -> str:
+        dockerfile = "# syntax = docker/dockerfile:1\n"  # Support BuildKit
+        dockerfile += "# Auto-generated by truss.build, do not edit!\n"
+        for command in self._commands:
+            dockerfile += command.serialize() + "\n"
+        return dockerfile
+
+    def apt_install(
+        self,
+        *packages: Union[str, List[str]],
```

Comment:
Optional/Opinion: is there a reason to support both variable args and list?

While it may seem more user friendly to have multiple and flexible options, I personally think it's less mental load to have exactly one right way of doing things and seeing that everywhere consistently, as opposed to having to make a decision each time knowing about "two forms".

Reply:
I am mostly trying to be API compatible with `modal` but we can have our own take here. Open to discussion.